core/Settings/UserSetting.php

```diff
@@ -34,9 +40,32 @@ public function __construct($name, $title, $userLogin = null)
         parent::__construct($name, $title);
 
         $this->setUserLogin($userLogin);
+    }
+
+    /**
+     * Returns `true` if this setting can be displayed for the current user, `false` if otherwise.
+     *
+     * @return bool
+     */
+    public function isReadableByCurrentUser()
+    {
+        return $this->isWritableByCurrentUser();
+    }
+
+    /**
+     * Returns `true` if this setting can be displayed for the current user, `false` if otherwise.
+     *
+     * @return bool
+     */
+    public function isWritableByCurrentUser()
+    {
+        if (isset($this->hasReadAndWritePermission)) {
+            return $this->hasReadAndWritePermission;
+        }
+
+        $this->hasReadAndWritePermission = Piwik::isUserHasSomeViewAccess();
```

Comment:
We do now check for permission only on demand. This could have triggered a DB query in tracker mode otherwise